### PR TITLE
[IMP] Expand env vars in config

### PR DESCRIPTION
**Pull request by lasley**
* Add optional environment variable expansion for the config file

### CODE REVIEW

**Comment by lmignon** [comment]
@lasley Thank you for this improvement! As stated by @Yajo an entry into the documentation for this new feature will be greatly appreciated.

**Comment by lasley** [comment]
Oops I knew I was forgetting something! Documentation updated

**Comment by lasley** [comment]
Coverage drop is due to more lines being added in the ReadMe it seems. I submitted #9 for that.

**Comment by lmignon** [comment]
@lasley IMO Variable expansion should be the default. A plus should be to keep a parameter to disable the expansion.

**Comment by sbidoul** [comment]
@lmignon @lasley are there security implications when enabling env var substitution?

**Comment by lasley** [comment]
@lmignon Making it the default sounds good to me - not being a core contributor to this repo, I was trying to make the change as silently as possible for the proposal.

@sbidoul Yeah there could be some security implications here, as with most things using environment vars. An example of a harmful act that could be performed with this is the using a Travis Secret env var in a test & dumping it to console. 

This would require a merge to master though, because the Travis secures are not exposed to forks. I'm sure some other dangerous things could be thought of as well, but that's the main one I watch out for in PRs with systems that include env expansion.

As with most security holes, the tradeoff is usability. In my specific instance, these env vars enhance usability of this application quite a bit in the context of Docker.

**Comment by sbidoul** [comment]
I prefer to keep the explicit activation of the substitution mode.

File: git_aggregator/main.py [hunk 1]
```diff
@@ -89,6 +89,16 @@ def get_parser():
         type=_log_level_string_to_int,
         nargs='?',
         help='Set the logging output level. {0}'.format(_LOG_LEVEL_STRINGS))
+
+    main_parser.add_argument(
+        '-e', '--expand-env',
```
Comment (lasley) [inline git_aggregator/main.py]: Per @pedrobaeza in https://github.com/Tecnativa/git-aggregator/commit/50467a388c419a65e0c5e8eeb5dca01b76742f8a#commitcomment-20682952:
> this is an incorrect use of parse arguments.

Comment (pedrobaeza) [inline git_aggregator/main.py]: Correct use is:
```
    main_parser.add_argument(
        '-e', '--expand-env',
        dest='expand_env',
        action='store_true',
        help='Expand environment variables in configuration file',
)
```

**Pull request merged**